pkg/cli/pod/pod_test.go

```diff
@@ -0,0 +1,299 @@
+									Status: corev1.ConditionTrue,
+								},
+							},
+						},
+					},
+				},
+			},
+			Namespace:   "default",
+			JobName:     "",
+			ExpectedErr: nil,
+			ExpectedOutput: `Name          Ready      Status         Restart  Age       
+my-pod        0/1        Running        0        0s`,
+		},
+		{
+			name: "Normal Case with jobName filter",
+			Response: &corev1.PodList{
+				Items: []corev1.Pod{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "my-pod",
+							Namespace: "default",
+							Labels: map[string]string{
+								v1alpha1.JobNameKey: "my-job1",
+							},
+							CreationTimestamp: metav1.Now(),
+						},
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "my-container",
+									Image: "nginx",
+								},
+							},
+						},
+						Status: corev1.PodStatus{
+							Phase: corev1.PodRunning,
+							Conditions: []corev1.PodCondition{
+								{
+									Type:   corev1.PodReady,
+									Status: corev1.ConditionTrue,
+								},
+							},
+						},
+					},
+				},
+			},
+			Namespace:   "default",
+			JobName:     "my-job1",
+			ExpectedErr: nil,
+			ExpectedOutput: `Name          Ready      Status         Restart  Age       
+my-pod        0/1        Running        0        0s`,
+		},
+		{
+			name: "Normal Case with queueName filter",
+			Response: &corev1.PodList{
+				Items: []corev1.Pod{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "my-pod1",
+							Namespace: "default",
+							Labels: map[string]string{
+								v1alpha1.QueueNameKey: "my-queue1",
+							},
+							CreationTimestamp: metav1.Now(),
+						},
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "my-container",
+									Image: "nginx",
+								},
+							},
+						},
+						Status: corev1.PodStatus{
+							Phase: corev1.PodRunning,
+							Conditions: []corev1.PodCondition{
+								{
+									Type:   corev1.PodReady,
+									Status: corev1.ConditionTrue,
+								},
+							},
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "my-pod2",
+							Namespace: "default",
+							Labels: map[string]string{
+								v1alpha1.JobNameKey:   "my-job2",
+								v1alpha1.QueueNameKey: "my-queue1",
+							},
+							CreationTimestamp: metav1.Now(),
+						},
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "my-container",
+									Image: "nginx",
+								},
+							},
+						},
+						Status: corev1.PodStatus{
+							Phase: corev1.PodRunning,
+							Conditions: []corev1.PodCondition{
+								{
+									Type:   corev1.PodReady,
+									Status: corev1.ConditionTrue,
+								},
+							},
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "my-pod3",
+							Namespace: "default",
+							Labels: map[string]string{
+								schedulingv1beta1.QueueNameAnnotationKey: "my-queue1",
```

Comment:
here is the test case

Reply:
I mean if both specified queue name and vcjob name, it will get an empty result, this case just set the queue name only.
And also your test case in server here doesn't execute the selector filter logic.